department-of-veterans-affairs/abd-vro
# MCP-2131: improve database auditing

## What was the problem?

Audit events track the progress of a claim though the routes but do not provide input/output information so it’s hard to tell what ultimately happens to the claim. We need a better way.

Associated tickets or Slack threads:

- [MCP-2131](https://amida.atlassian.net/browse/MCP-2131)

## How does this fix it?[^1]

- Events originating from the same endpoint request have a correlation ID that links them together

- Events details contain no PII but have identifying information relating to the claim such as claimId, collectionId, and veteran Icn, 

- There is roughly one event per arrow in this diagram: [VRO v2 Roadmap · department-of-veterans-affairs/abd-vro Wiki](https://github.com/department-of-veterans-affairs/abd-vro/wiki/VRO-v2-Roadmap)

- Each event describes the outcome of an operation without mentioning PII. For example, if the claim is in scope, if there is sufficient evidence, if an exam was ordered, etc.

## How to test this PR
- Process Mas Claims under different scenarios
- The returned object will give the correlation id for each request
- Inspect the database by querying the audit_event table with the given correlation ID


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
